src/notebooksyncagent.cpp

```diff
@@ -155,223 +194,242 @@ void NotebookSyncAgent::startQuickSync(mKCal::Notebook::Ptr notebook,
+        // we have a hash from resource href-uri to resource info (including etags).
+        QMultiHash<QString, Reader::CalendarResource> map = report->receivedCalendarResources();
+        QHash<QString, QString> remoteHrefUriToEtags;
+        Q_FOREACH (const QString &href, map.keys()) {
+            if (!href.contains(m_remoteCalendarPath)) {
+                LOG_WARNING("href does not contain server path:" << href << ":" << m_remoteCalendarPath);
+                emitFinished(Buteo::SyncResults::INTERNAL_ERROR, "unable to calculate remote resource uids");
+                return;
+            }
+            QList<Reader::CalendarResource> resources = map.values(href);
+            for (int i = 0; i < resources.size(); ++i) {
+                remoteHrefUriToEtags.insert(href, resources[i].etag);
+                break; // only one etag per resource is allowed.
+            }
         }
 
-        // store the base recurring event into the in-memory calendar
-        memoryCalendar->addIncidence(exportableIncidence);
+        // calculate the local and remote delta.
+        if (!calculateDelta(KDateTime(m_changesSinceDate),
+                            remoteHrefUriToEtags,
+                            &m_localAdditions,
+                            &m_localModifications,
+                            &m_localDeletions,
+                            &m_remoteAdditions,
+                            &m_remoteModifications,
+                            &m_remoteDeletions)) {
+            emitFinished(Buteo::SyncResults::INTERNAL_ERROR, "unable to calculate sync delta");
+            return;
+        }
 
-        // now create the persistent occurrences in the in-memory calendar
-        Q_FOREACH (KCalCore::Incidence::Ptr instance, instances) {
-            // We cannot call dissociateSingleOccurrence() on the MemoryCalendar
-            // as that's an mKCal specific function.
-            // We cannot call dissociateOccurrence() because that function
-            // takes only a QDate instead of a KDateTime recurrenceId.
-            // Thus, we need to manually create an exception occurrence.
-            KCalCore::Incidence::Ptr exportableOccurrence(exportableIncidence->clone());
-            exportableOccurrence->setCreated(instance->created());
-            exportableOccurrence->setRevision(instance->revision());
-            exportableOccurrence->clearRecurrence();
-            exportableOccurrence->setRecurrenceId(instance->recurrenceId());
-            exportableOccurrence->setDtStart(instance->recurrenceId());
+        // Note that due to the fact that we update the ETAG and URI data in locally
+        // upsynced events during sync, those incidences will be reported as modified
+        // during the next sync cycle (even though the only changes may have been
+        // that ETAG+URI change).  Hence, we need to fetch all of those again, and
+        // then manually check equivalence (ignoring etag+uri value) with remote copy.
```

Comment:
You should be able to setLastModified() on the incidence with the previously set last-modified timestamp to avoid mkcal reporting it as modified in the next sync cycle.